### PR TITLE
Add filters popover to the users admin table

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56262
https://mattermost.atlassian.net/browse/MM-56589
https://mattermost.atlassian.net/browse/MM-56587
https://mattermost.atlassian.net/browse/MM-56588

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/index.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+type FiltersState = Partial<Pick<AdminConsoleUserManagementTableProperties, 'filterRole' | 'filterStatus'>>;
+
+interface Props {
+    filterRole: AdminConsoleUserManagementTableProperties['filterRole'];
+    filterStatus: AdminConsoleUserManagementTableProperties['filterStatus'];
+}
+
+export function SystemUsersFilterPopover(props: Props) {
+    const dispatch = useDispatch();
+
```
Comment (devinbinnie) [inline webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/index.tsx:30]: nit: a lot of unnecessary spacing here

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/system_users/system_users_filters_popover/index.tsx:30]: fixed

**Review by devinbinnie — APPROVED** [review]

**Pull request merged**